### DISCUSSION TITLE
Helm chart user code deployments.

### DISCUSSION THREAD

**Discussion by andrescolodrero** [dagster-io/dagster#19070] [Q&A]
How can we separate, in a Helm Chart, the provision of a Dagster server (web, daemon, etc) from user-code locations?
The idea is  to have separate repos, one for admins, other for users to deploy their code.

**Comment by hnykda** [comment]
I came to the conclusion that it's not possible.

https://github.com/dagster-io/dagster/discussions/14382

I find it really surprising! We wanted simple:

1. dagster coortinator/orchestrator/webserver deployment, no user code
2. repoA as a user deployment location connecting to the server
3. repoB  as a user deployment location connecting to the server

Seems that this is not possible and you have to have single `dagster-user-deployments` helm deployment with all locations. You either have to deploy all three together or deploy [1 without subgraphs and then the two together](https://docs.dagster.io/deployment/guides/kubernetes/customizing-your-deployment#separately-deploying-dagster-infrastructure-and-user-code).

I find this very limiting - it means you have to colocate deployment of all pipelines/teams in one file 🙄

**Reply by ion-elgreco** [reply]
@andrescolodrero  @hnykda We have solved this using a custom CLI: https://github.com/ASML-Labs/dagster-uc

This means you can have separate project repos, that will deploy their code location to the same dagster instance.
